src/app/workflow/dag/wdl-viewer/state/wdl-viewer.model.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * Defines types for the response of Pipeline Builder library
+ */
+export interface WdlViewerPipeline {
+  status: boolean;
+  message: string;
+  model: Array<any>;
+  actions: Array<any>;
+  id?: number;
```

Comment:
This is getting nit-picky, but it looks like the id isn't actually part of the response, but something you add on -- the id of the workflow version. I was going to comment in the `wdl-viewer.service.update()` method that your code might fail because if the id is not set (because it's declared optional here), but it looks like you always ensure the id is set -- but you're not letting the compiler do the checking for you.

You don't need to over-architect this, so I suggest at a bare minimum commenting and/or renaming the id field to clearly indicate that that it is not part of the response.

More elaborately, you could try using [intersection types](https://www.typescriptlang.org/docs/handbook/advanced-types.html), where WdlViewerPipeline is the response, and you have a new type that is an intersection of WdlViewerPipeline and id, where is not optional.

Reply:
Regarding the intersection type, does this mean that the store and query should extend an intersection of `WdlViewerPipeline` and `id`?

Reply:
Looks like it would have to, and may not be worth the effort.

To be clear, my concern was that you have a property, id, that is declared optional, yet your code relies on it always being present, and you don't do null checks.

OTOH, the code is not that complex, and it's unlikely there will be any additional users of this particular code, so it's probably overkill to spend too much time on this.